swift/example_code/s3/presigned-urls/Sources/entry.swift

```diff
@@ -0,0 +1,503 @@
+        let (_, response) = try await URLSession.shared.upload(for: request, fromFile: source)
+        guard let response = response as? HTTPURLResponse else {
+            throw TransferError.uploadError("No response from Amazon S3")
+        }
+        
+        if response.statusCode != 200 {
+            throw TransferError.uploadError(
+                "Upload failed with status code: \(response.statusCode)"
+            )
+        } else {
+            print("File uploaded to \(source.absoluteString)")
+        }
+    }
+
+    /// Use the specified `URLRequest` to download a file.
+    ///
+    /// - Parameters:
+    ///   - request: The presigned URLRequest to perform.
+    ///   - dest: The file system URL to relocated the fully downloaded file to.
+    ///
+    /// - Throws: `TransferError.downloadError`, `TransferError.writeError`,
+    ///   `TransferError.fileMoveError`
+    ///
+    /// The file is first downloaded to the user's Downloads directory, then
+    /// it's copied to the destination URL.
+    func httpFetchRequest(request: URLRequest, dest: URL) async throws {
+        let (fileURL, response) = try await URLSession.shared.download(for: request)
+        guard let response = response as? HTTPURLResponse else {
+            throw TransferError.downloadError("No response from Amazon S3")
+        }
+        
+        // If the download was successful, move the file to the destination.
+        
+        if response.statusCode == 200 {
+            do {
+                try FileManager.default.moveItem(at: fileURL, to: dest)
+                print("File saved as \(dest.lastPathComponent)")
+            } catch {
+                throw TransferError.fileMoveError
+            }
+        } else {
+            print("ERROR: Download failed with HTTP status code: \(response.statusCode)")
+            throw TransferError.downloadError(
+                "Download failed with HTTP status code: \(response.statusCode)"
+            )
+        }
+    }
+
+    // -MARK: - Asynchronous main code
+    
+    /// Called by ``main()`` to run the bulk of the example.
+    func runAsync() async throws {
+        switch direction {
+        case .up:
+            if multiPart == false {
+                try await uploadFile(sourcePath: source, bucket: bucket,
+                                     key: dest)
+            } else {
+                try await uploadFileMultipart(sourcePath: source, bucket: bucket,
+                                              key: dest)
+            }
+        case .down:
+            try await downloadFile(bucket: bucket, key: source,
+                                   destPath: dest)
+        }
+    }
+}
+
+// -MARK: - Entry point
+
+/// The program's asynchronous entry point.
+@main
+struct Main {
+    static func main() async {
+        let args = Array(CommandLine.arguments.dropFirst())
+
+        await SDKLoggingSystem().initialize(logLevel: .critical)
```

Comment:
Are these the default logLevels now?

Reply:
The defaults are .error, I think, but I'm removing this logging stuff anyway. I'd planned to leave it but now that I think about it slightly more separated from the work, I think it should go.